### ISSUE TITLE
Rate limit trading

### ISSUE THREAD

**Issue by hasteech** [endless-sky/endless-sky#1678]
It's a bit too easy to grind cash with hundreds of freighters stripped for cargo.  It also doesn't make a lot of in-game sense to head to some tiny outpost at the edge of the galaxy and unload 400k+ of industrial and pick up another 400k+ of electronics.  I think the simplest thing might be to use the dynamic economy weights to both scale prices and set a limit on the amount of each good we can buy or sell.  There are some interesting options for lore-based balance as well, such as having high volume/low profit/low risk areas (core settlements), low volume/low profit/low risk areas (poor outlying outposts), and low volume/high profit/high risk areas (pirated areas).

**Comment by jafdy** [comment]
A dynamic economy has been in exsistantce since the start of 0.9

**Comment by hasteech** [comment]
Yes.  That's why I suggested using its to limit the quantity of goods in addition to the price.

**Comment by KiLEdEnNis** [comment]
that would be like saying "make human fleets less frequent as I killed 40k ships"
I don't think this should be added, but it's a good suggestion

**Comment by hasteech** [comment]
It's more like having the traders on the planets say "I only 100 metal on hand, and I only have 100k credits to purchase with, but I might have more if you come back next week."  It's realistic and addresses the exponential star barge problem.

**Comment by KiLEdEnNis** [comment]
but it would be annoying, personally I'd say ghameplay over realism on this one (the economy can already inflate/fluctuate when buying/selling large quantities)

**Comment by LocalGod79** [comment]
@hasteech have you actually used the exponential Star Barge stripped for cargo?  did something happen to Freighter raiders (so they no longer spawn)?

**Comment by hasteech** [comment]
I say exponential star barge, but I upgrade to freighters, haulers of some sort, and then bulk freighters.  There are no raiders in Hai space, and cargo warships avoid raiders as well.  I think there's a strategy for fleeing the raiders as well, but I've only done that on a modded game and haven't tried it in vanilla yet.

**Comment by Amazinite** [comment]
I remember a while back MZ saying that he might add Unfettered raiders in Hai space, and they could be used in Wanderer space now as well. That would certainly make Hai space not so much of an OP safe haven for freighters.

**Comment by hasteech** [comment]
It would help, but there are other ways to avoid the raiders.  A giant stack of steering+turret focused freighters (behemoths are probably best for this) will crush everything that comes close, and fast steering+slow speed means a tight cluster.  I just tested fleeing raiders and it seems to work in vanilla, albeit a bit more slowly.  I used a fast scram drive bounder and 500 bulk freighters.  The bounder is maxed for steering, speed, then shield regen, and the freighters are maxed for steering then cargo.  Depart, fleet holds, jump, outrun raiders until they stop pursuit, regen shields, jump, repeat to destination, outrun raiders until they jump away or are defeated, gather fleet, and land.  There are probably other strategies.  Raiders aren't a great way of discouraging massive freighter fleets.
